torch/csrc/jit/graph_executor.cpp

```diff
@@ -444,6 +445,7 @@ struct GraphExecutorImpl {
   void runOptimization(std::shared_ptr<Graph>& graph, const ArgumentSpec& spec) {
     EliminateDeadCode(graph);
     EliminateCommonSubexpression(graph);
+    ConstantPooling(graph);
```

Comment:
As I mentioned, this should really happen at the very end of optimization, because other passes can introduce unnecessary constants (e.g. `PeepholeOptimize` can replace `mm` + `add` with `addmm` inserting ones for `alpha` and `beta`).

Reply:
I think we should run it before UnrollLoops so that the size of the loops is minimized. In the upcoming diff where constants are no longer instructions, it will be run as a pass in the interpreter to guarantee that all constants are at folded.

Reply:
I will update this in the forthcoming diff